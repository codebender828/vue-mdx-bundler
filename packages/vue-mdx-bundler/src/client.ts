import * as vue from 'vue'

function getMDXComponent(code: string, globals?: Record<string, unknown>) {
  const scope = { vue, ...globals }
  // eslint-disable-next-line
  const fn = new Function(...Object.keys(scope), code)
  return fn(...Object.values(scope))
}

export { getMDXComponent }
