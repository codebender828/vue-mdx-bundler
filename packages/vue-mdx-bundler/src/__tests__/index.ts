import { getMDXComponent } from '../client'
import { render } from '../../tests/test-utils'
import { bundleMDX } from '../index'

test('returns code and frontmatter', async () => {
  const mdxSource = `
---
title: Example Post
published: 2021-02-13
description: This is some meta-data
---
# This is the title

Here's **neat** demo:
`.trim()

  const result = await bundleMDX(mdxSource, {
    files: {},
  })

  // const MdxContent = `
  // import * as vue from "vue";
  // export const MdxContent = () => { ${result.code} };
  // export const frontmatter = ${JSON.stringify(result.frontmatter)}; `.trim()
  console.log(result.code)

  const MdxComponent = getMDXComponent(result.code)

  const { container } = render({
    template: '<MdxComponent></MdxComponent>',
    components: { MdxComponent },
  })

  expect(container).toMatchSnapshot()

  expect(result.frontmatter).toEqual({
    description: 'This is some meta-data',
    published: new Date('2021-02-13T00:00:00.000Z'),
    title: 'Example Post',
  })
})

test('files is optional', async () => {
  await expect(bundleMDX('hello')).resolves.toBeTruthy()
})
