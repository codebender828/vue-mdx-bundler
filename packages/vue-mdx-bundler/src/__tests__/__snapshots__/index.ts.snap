// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`returns code and frontmatter 1`] = `
<div>
  
  <h1>
    This is the title
  </h1>
  

  <p>
    Here's 
    <strong>
      neat
    </strong>
     demo:
  </p>
  
</div>
`;
